dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/OrgUnitJdbcPredicate.java

```diff
@@ -0,0 +1,168 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors 
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export;
+
+import static org.hisp.dhis.common.IdentifiableObjectUtils.getIdentifiers;
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.ALL;
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.CAPTURE;
+import static org.hisp.dhis.user.CurrentUserUtil.getCurrentUserDetails;
+
+import java.util.Set;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.common.OrganisationUnitSelectionMode;
+import org.hisp.dhis.commons.util.SqlHelper;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class OrgUnitJdbcPredicate {
+
+  public static void buildOrgUnitModeClause(
+      StringBuilder sql,
+      Set<OrganisationUnit> orgUnits,
+      OrganisationUnitSelectionMode orgUnitMode,
+      MapSqlParameterSource sqlParameters,
+      String tableAlias) {
+
+    switch (orgUnitMode) {
+      case DESCENDANTS -> addOrgUnitDescendantsCondition(sql, orgUnits, sqlParameters, tableAlias);
+      case CHILDREN -> addOrgUnitsChildrenCondition(sql, orgUnits, sqlParameters, tableAlias);
+      case SELECTED -> {
+        sql.append(tableAlias).append(".organisationunitid in (:orgUnits) ");
+        sqlParameters.addValue("orgUnits", getIdentifiers(orgUnits));
+      }
+      case ALL, CAPTURE, ACCESSIBLE -> {
+        // these modes don't accept org units, so skip predicate
+      }
+    }
+  }
+
+  public static void buildOwnershipClause(
+      StringBuilder sql,
+      OrganisationUnitSelectionMode orgUnitMode,
+      MapSqlParameterSource sqlParameters,
+      Set<OrganisationUnit> effectiveSearchOrgUnits,
+      Set<OrganisationUnit> captureScopeOrgUnits,
```

Comment:
These parameters are not coming from the user input, could we calculate them inside the method?